v0x04/match_fields.py

```diff
@@ -39,10 +39,16 @@ def as_of_tlv(self):
             value = int(self.value)
             mask = None
             oxm_hasmask = False
+            value = value | VlanId.OFPVID_PRESENT
         except ValueError:
-            value, mask = map(int, self.value.split('/'))
-            oxm_hasmask = True
-        value = value | VlanId.OFPVID_PRESENT
+            try:
+                value, mask = map(int, self.value.split('/'))
+                oxm_hasmask = True
+                value = value | VlanId.OFPVID_PRESENT
+            except ValueError:
+                mask = None
+                oxm_hasmask = False
+                value = VlanId.OFPVID_NONE
```

Comment:
Scenario 3: Matching only packets with a VLAN tag regardless of its value: `any`
`value = VlanId.OFPVID_NONE`
Scenario 2: Matching only packets without a VLAN tag: `untagged`
`value = VlanId.OFPVID_PRESENT`

Reply:
Changed.